src/realm/table_view.cpp

```diff
@@ -416,6 +396,12 @@ void TableView::limit(LimitDescriptor lim)
     do_sync();
 }
 
+void TableView::filter(FilterDescriptor filter)
+{
+    m_descriptor_ordering.append_filter(std::move(filter));
+    do_sync();
```

Comment:
Shouldn't it only attempt full do_sync if limit is present? Otherwise just do the filter and nothing else.

Reply:
I just wanted to handle it in the same way as the other functions.